cmd/controller-manager/app/controller-manager.go

```diff
@@ -67,8 +67,11 @@ member clusters and does the necessary reconciliation`,
 		},
 	}
 
-	opts.AddFlags(cmd.Flags())
-	cmd.Flags().BoolVar(&verFlag, "version", false, "Prints the Version info of controller-manager")
+	opts.AddFlags(pflag.CommandLine)
+	pflag.CommandLine.BoolVar(&verFlag, "version", false, "Prints the Version info of controller-manager")
```

Comment:
Why is this line changing?

Reply:
two reasons:
- pflag.CommandLine is one FlagSet obj, see [this](https://github.com/kubernetes-sigs/federation-v2/blob/master/vendor/github.com/spf13/pflag/flag.go#L1200), and cmd.Flags() will new another FlagSet obj., see [this](https://github.com/kubernetes-sigs/federation-v2/blob/master/vendor/github.com/spf13/cobra/command.go#L1287-L1297)
- and "k8s.io/apiserver/pkg/util/flag" chooses pflag.CommandLine as default, see [this](https://github.com/kubernetes-sigs/federation-v2/blob/master/vendor/k8s.io/apiserver/pkg/util/flag/flags.go#L49)

but, if you don't want to use "k8s.io/apiserver/pkg/util/flag", there will be another way:
```
import "flag"

opts.AddFlags(cmd.Flags())
cmd.Flags().AddGoFlagSet(flag.CommandLine)
```